### ISSUE TITLE
Interfaces

### ISSUE THREAD

**Issue by Xerosigma** [Yonaba/Lua-Class-System#9]
Do you think it would be possible to implement interfaces of some sort that allow multiple inheritance?

Long story short maybe something like:

```
IPublisher = LCS.class.interface({})

function IPublisher:Methods()
    return
    {
        subscribe = Subscribe,
        unsubscribe = Unsubscribe,
    }
end

function IPublisher:Subscribe( sender, args ) end
function IPublisher:Unsubscribe( sender, args ) end
```

Then on class construction check the objects and throw errors if a method was not implemented?

Implementation could look like so:

```
SomePublisher = LCS.class({ implements{ IPublisher, ISomeOtherInterface } })
```

I think this would be a great addition to your library, do you think it's feasible?

Thank you.

**Comment by Yonaba** [comment]
It seems the point here is not really Interfaces, but Multiple Inheritance, as a new feature ?
Well, this is something I would not want to look into. Moslty because it is not much needed as feature, from my own experience, and moslty because in most cases you might think you need it, it is possible to find a workaround.

Second, multiple inheritance can lead to awkward situations especially when different superclass implements the same method. When a class inherits from those superclass, what would be final method the derived class will implement ? (see the [diamon problem](http://en.wikipedia.org/wiki/Multiple_inheritance#The_diamond_problem)).

In a nutshell, I have considered this, but I did not implement it. Multiple Inheritance is covered in [PiL](http://www.lua.org/pil/16.3.html), and some librairies out there implement it, but I wanted to stick to the single inheritance scheme for this library.

At least, we have mixins. Not implemented yet, but i'll be looking into it.

**Comment by Xerosigma** [comment]
:+1: For mixins, that's good enough.
